plugins/source/terraform/plugin/plugin.go

```diff
@@ -0,0 +1,37 @@
+package plugin
+
+import (
+	"github.com/cloudquery/cloudquery/plugins/source/terraform/client"
+	"github.com/cloudquery/cloudquery/plugins/source/terraform/services"
+	"github.com/cloudquery/plugin-sdk/plugins"
+	"github.com/cloudquery/plugin-sdk/schema"
+)
+
+const exampleConfig = `
+backends:
+  - name: mylocal # local backend
+    local:
+      path: ./examples/terraform.tfstate
+#  - name: myremote # s3 backend
+#    s3:
+#      bucket: tf-states
+#      key: terraform.tfstate
+#      region: us-east-1
+#      role_arn: ""
+`
```

Comment:
I would suggest changing this config format to something like this:

```
backends:
  - name: mylocal # local backend
    type: local
    params:
      path: ./examples/terraform.tfstate
#  - name: myremote # s3 backend
#    type: s3
#    params: 
#      bucket: tf-states
#      key: terraform.tfstate
#      region: us-east-1
#      role_arn: ""
```

Each "type" can then specify its own set of parameters. This way, we don't have to rely on the presence / absence of keys to tell what kind of remote backend it is. We can also more easily check whether a given "type" is supported by the current version of the code, and error out if it is not. WDYT?

Reply:
This will complicate things, we'll basically need to have a copy of specreader-style code in the plugin. Which is not pretty. (The old `",inline"` thing doesn't work because we're supposed to use json tags/decoder even if we're using yaml config.)

Reply:
Hmm, if we want to keep things simple, can we use a `map[string]interface{}` type for the params, then place the values into appropriate structs with some custom logic?

Reply:
Fixed